### ISSUE TITLE
Decide what to do about IDNA encoding

### ISSUE THREAD

**Issue by mithrandi** [twisted/txacme#125]
Cryptography no longer IDNA encodes names for us, should we be doing this instead?

**Comment by markrwilliams** [comment]
> Cryptography no longer IDNA encodes names for us, should we be doing this instead?

Yes - perhaps this can begin by making `ICertificateStore` impose an encoding for `server_name`?  This should be pretty easy for `DirectoryStore`.
